File: lib/db/repo/src/info_log_repo.dart
/*
 * Copyright (c) 2022.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

part of db_repo;

class _InfoLogRepo {
  Future<FError> insert(InfoLogEntity e) async {
    final sql = '''
INSERT INTO info_log
  (id, uid, user_id, partner_id, xid,
   svc_name, svc_parent, message, severity, path,
   function, data, created_at)
VALUES
  (@id, @uid, @user_id, @partner_id, @xid,
   @svc_name, @svc_parent, @message, @severity, @path, 
   @function, @data, @created_at)
''';

    final pars = {
      'id': e.id,
      'uid': e.uid,
      'user_id': e.userId,
      'partner_id': e.partnerId,
      'xid': e.xid,
      'svc_name': e.svcName,
      'svc_parent': e.svcParent,
      'message': e.message,
      'severity': e.severity,
      'path': e.path,
      'function': e.function,
      'data': e.data,
      'created_at': e.createdAt,
    };

    return await app.db.execute(sql: sql, pars: pars);
  }
}

File: clog-go/db/repo/info-log-repo.go
/*
 * Copyright (c) 2023.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

package repo

import (
	"clog/app"
	"clog/db/entity"

	"github.com/andypangaribuan/project9/core/db"
)

type srInfoLog struct {
	repo *db.Repo[entity.InfoLog]
}

func (slf srInfoLog) new() *srInfoLog {
	slf.repo = new(db.Repo[entity.InfoLog])
	slf.repo.DbInstance = app.Db
	slf.repo.PrintUnsafeErr = !app.Env.AppEnv.IsProd()

	slf.repo.TableName = "info_log"
	slf.repo.SetInsertColumnNames(`
		id, uid, user_id, partner_id, xid,
		svc_name, svc_version, svc_parent, message, severity,
		path, function, data, created_at`)
	slf.repo.InsertParamSign = slf.repo.GenerateParamSigns(slf.repo.InsertColumnNames)

	return &slf
}

func (slf *srInfoLog) Insert(e entity.InfoLog) error {
	return slf.repo.Insert(nil,
		e.Id, e.Uid, e.UserId, e.PartnerId, e.Xid,
		e.SvcName, e.SvcVersion, e.SvcParent, e.Message, e.Severity,
		e.Path, e.Function, e.Data, e.CreatedAt,
	)
}
